docs/source/articles/decorate-and-write-tests.html.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Decorate & Write Tests
+created_at: 2020/01/23
+excerpt: How to decorate Workarea tests and write your own application tests
+---
+
+# Decorate & Write Tests
+
+From your application, you can decorate the tests that ship with Workarea, and you can write your own application tests.
+See [Testing Concepts, Tests & Decorators](/articles/testing-concepts.html#tests-amp-decorators) for more info, including an explanation of when to decorate vs when to write your own.
+
+This doc explains how to [Decorate a Workarea Test Case](#decorate-a-workarea-test-case), including the special case [Skip a Workarea Test](#skip-a-workarea-test).
+
+This doc also covers how to [Write an Application Test Case](#write-an-application-test-case), and when doing so, the following recipes may also be useful:
+
+* [Change Configuration within a Test](#change-configuration-within-a-test)
+* [Change Locale within a Test](#change-locale-within-a-test)
+* [Change Time within a Test](#change-time-within-a-test)
+
+
+## Decorate a Workarea Test Case
```

Comment:
There's one point of guidance not mentioned here, I'm not sure exactly what the best explanation would be. The idea is that decorating tests to add tests is almost always better done in a new test file, like `test/models/workarea/payment/my_site_credit_card_integration_test.rb` It makes it way easier to write the new test in terms of running and traces. It also gives you the ability to do only the test setup you need for your actual tests instead of the setup built for the ones in the base test class.

Reply:
Ben makes a good point. The guidance I usually give around this is that a developer should only decorate a test if the host app breaks or invalidates the core test. Otherwise it is best practice to create new test files.